common/persistence/sql/sqlplugin/mysql/visibility.go

```diff
@@ -82,7 +82,9 @@ var (
 func buildOnDuplicateKeyUpdate(fields ...string) string {
 	items := make([]string, len(fields))
 	for i, field := range fields {
-		items[i] = fmt.Sprintf("%s = VALUES(%s)", field, field)
+		// This line is to ensure that no update occurs (for any column) if the version is behind the saver version.
+		items[i] = fmt.Sprintf("%v = IF(%v < VALUES(%v), VALUES(%v), %v)",
+		 field, sqlplugin.VersionColumnName, sqlplugin.VersionColumnName, field, field)
```

Comment:
We need to make sure the `version` column is the last to update. Otherwise, the condition will fail after the `version` column is updated.

Reply:
Yes. I put that in a different section, but I think we can get away with just ordering the field in the struct. I'll make a test to ensure the order comes back correct, but golang also guarantees field order matches declaration (except for embedding) in reflection.